force-app/main/default/lwc/newServiceSchedule/newServiceSchedule.js

```diff
@@ -49,13 +56,74 @@ export default class NewServiceSchedule extends LightningElement {
     }
 
     @api reportValidity() {
-        // TODO: add validity checks for enddatetime > startdatetime, frequency != null, dow != null when freq = weekly, etc.
-        return [...this.template.querySelectorAll("lightning-input-field")].reduce(
-            (validSoFar, inputField) => {
-                return validSoFar && inputField.reportValidity();
-            },
-            true
-        );
+        let errMessages = [];
+
+        let isFormValid = [
+            ...this.template.querySelectorAll("lightning-input-field"),
+        ].reduce((validSoFar, inputField) => {
+            return validSoFar && inputField.reportValidity();
+        }, true);
+
+        let datesValid = this.dateFields.start.value < this.dateFields.end.value;
+
+        let hasEndCondition = this.validateServiceScheduleOnOrAfter();
+
+        let hasDayOfWeek =
+            this.picklistFields.frequency.value === WEEKLY
+                ? this.picklistFields.daysOfWeek.value
+                : true;
+
+        // input-fields handle their own error display, so no additional error message needed for !isFormValid
+        if (!datesValid) {
+            errMessages.push(
+                format(START_BEFORE_END_LABEL, [
+                    this.dateFields.start.label,
+                    this.dateFields.end.label,
+                ])
+            );
+        }
+
+        if (!hasEndCondition) {
+            errMessages.push(NO_END_LABEL);
+        }
+
+        if (!hasDayOfWeek) {
+            errMessages.push(DAY_REQUIRED_LABEL);
+        }
+
+        this.errorMessage = errMessages.join("\n");
+
+        this.isValid = isFormValid && datesValid && hasEndCondition && hasDayOfWeek;
```

Comment:
I like how you broke all of these conditions down, it is easy to follow what is required with this one line here.